src/main/kotlin/com/tailoredapps/gradle/localize/Localize.kt

```diff
@@ -111,54 +123,65 @@ class Localize {
-        if (checkLines == actualLines)
+        if (checkLines == actualLines) {
             return null
+        }
 
         val patch = DiffUtils.diff(checkLines, actualLines)
         val diff =
             DiffUtils.generateUnifiedDiff(expectedFileName, actualFileName, checkLines, patch, 3)
         return diff.joinToString("\n")
     }
 
-    private fun getStringsXmlFileOrThrow(
-        localizationPath: File,
-        localizationIdentifier: String?
-    ): File {
-        val valuesDirectory = File(
-            localizationPath,
-            if (localizationIdentifier == null) "values" else "values-$localizationIdentifier"
-        ).also { directory ->
-            if (directory.exists().not()) {
-                if (directory.mkdirs().not()) {
-                    throw IOException("Could not create directory (or parent directories for) ${directory.absolutePath}. This directory was defined as 'localizationPath' to save the strings.xml files in.")
+    private fun getStringsXmlFileOrThrow(localizationPath: File, localizationIdentifier: String?): File {
+        val valuesDirectory =
+            File(
+                localizationPath,
+                if (localizationIdentifier == null) "values" else "values-$localizationIdentifier"
+            ).also { directory ->
+                if (directory.exists().not()) {
+                    if (directory.mkdirs().not()) {
+                        throw IOException(
+                            "Could not create directory (or parent directories for) ${directory.absolutePath}. This directory was defined as 'localizationPath' to save the strings.xml files in."
+                        )
+                    }
                 }
             }
-        }
         return File(valuesDirectory, "strings.xml").also { file ->
             if (file.exists().not()) {
                 if (file.createNewFile().not()) {
-                    throw IOException("Could not create file ${file.absolutePath}. The location of this file was defined as 'localizationPath' to save the strings.xml files in.")
+                    throw IOException(
+                        "Could not create file ${file.absolutePath}. The location of this file was defined as 'localizationPath' to save the strings.xml files in."
+                    )
                 }
             }
         }
     }
 
+    @VisibleForTesting
+    internal fun LocalizationSheetParser.ParsedSheet.abortLocalizationOnKeyDuplicates() {
+        worksheets.forEachParallel { worksheet ->
+            val duplicates = worksheet.entries
+                .mapNotNull { it.identifier[LocalizationSheetParser.Platform.Android] }
+                .groupBy { it }
+                .filter { it.value.size > 1 }
 
-}
\ No newline at end of file
+            if (duplicates.isNotEmpty()) {
+                throw IllegalStateException("Duplicates found for the following key(s): ${duplicates.keys.joinToString(", ") { "\"$it\"" }}. Localization update aborted!")
```

Comment:
As we're iterating through different worksheets here, and the (current) use-case for that is also that there are multiple worksheets for multiple configurations, it might be worth to print the worksheet name in the exception as well?

Reply:
done